lib/fs/include/lib/fs.h

```diff
@@ -41,6 +50,8 @@ struct dirent {
 typedef struct filehandle filehandle;
 typedef struct dirhandle dirhandle;
 
+
+status_t fs_format_device(const char *fsname, const char *device, const void *args);
```

Comment:
should follow the convention and use the __NONNULL stuff here. in this case it's at least __NONNULL((1)) since the first arg is not optional.